### DISCUSSION TITLE
SingleCycleUniqueEvent scheduled but callback function not called

### DISCUSSION THREAD

**Discussion by redhawksanji** [sparcians/map#486] [Q&A]
Hi, 

I was using a SingleCycleUniqueEvent A to schedule the event in the same cycle in our model. The event A was called from another Payload event B, both in the Tick phase. There was no specific ordering mentioned between the events. I saw that my event A was never getting called. I did not have any cancellations for event A. What would be the reason that the event was scheduled but callback function never got called?

If I order the event A after event B, the callback function gets called. Doesn't the ordering happen automatically during compile time for such scenarios or do I have to specify the ordering between all the events?

**Comment by klingaard** [comment]
Hmmmm... that's a weird one.  Let me see if I can I fully comprehend the issue.

You have a `PayloadEvent` called B that when called in the `Tick` phase schedules `SingleCycleEvent` A, also in the `Tick` phase for 0-cycle offset.  But the scheduling of the `SingleCycleEvent` A never happens.  

But when you add a precedence rule: `ple_event_b >> sce_event_a` the event _will be_ scheduled?  I'm not sure why that would make a difference (from looking at the code), so that's very odd.

Couple of things to try out.  
1. You can use `gdb` to step through the `schedule` call to `sce_event_a` to see why it's ignoring you.  
2. You can use `-l scheduler debug sched.out --debug-on <rough cycle when it happens>` followed by `less -R sched.out` and see if the schedule call is even getting to the Scheduler.
